chatbot.py

```diff
@@ -35,9 +35,12 @@ def message_to_bot(H, clf, learn_response):
         B = "I will certainly help you with that."
         learn_response = 0
         return B, learn_response
-    if H.lower() == "bye" or H.lower() == "bye." or H.lower() == "bye!":   # empty input
+    if H.lower() == "bye" or H.lower() == "bye." or H.lower() == "bye!":   
```

Comment:
Hmm, would something like this be better?

```python3
# option 1
BYE_MESSAGES = ["bye", "bye.", "bye!"]
if H.lower() in BYE_MESSAGES:
    ...

# option 2
if H.lower().startswith("bye"):
    ...
```

Reply:
@athityakumar how about the following?
```
if "bye" in H.lower():   
    ...
```
This would also handle cases when the user types "Okay bye" or something like that. How does that sound?

Reply:
Hmm, but that'll also have edge cases (though I don't think it's very likely that someone will use it): words like lobbyer will also get considered as a bye phrase.

Reply:
@vishakha-lall - Thoughts?

Reply:
Yes, that's true. So should I go for the option 2 you suggested above then?

Reply:
Let's wait for @vishakha-lall to tell her opinion as well. I think we can go with option 2 like

```python3
if "bye" in H.lower().split(" "): # check in words within H
    ...
```

Reply:
Hmm, I agree with @athityakumar's latest suggestion. That should be a good implementation.

Reply:
I tried to test the code using the latest suggestion, but it fails to recognize the cases where "bye!" or "bye." is used. I tried a few more different implementations, but none of them seems to work well. Could you help me out?

Reply:
@asthacs I think this issue is coming since we have no text pre-processing in place which would filter out elements like punctuations, stop words etc. from the input text. That would solve the scenarios that you're facing here. Also, @athityakumar's suggestion would do a fair job in most cases but since it can run into edge cases, I'd suggest we look for some more approaches before we close this.

Reply:
Ideally, sentence/word similarities would be the way to go about this issue. However, that is far beyond the scope of this issue but we should add it to our 'backlog' :P @athityakumar @vishakha-lall

Reply:
@janakrajchadha so what changes should I make on my end?

Reply:
You can go with the latest suggestion by @athityakumar

Reply:
@janakrajchadha can you create a new issue around your suggestion - to implement word/sentence similarities for this case.